test/memory-cache.test.js

```diff
@@ -0,0 +1,80 @@
+import { expect } from '@bundled-es-modules/chai';
+import * as hanbi from 'hanbi';
+import { MemoryCache } from '../src/lib/memory-cache.js';
+
+describe('memory cache', function () {
+  it('should call provided function when cache miss', function () {
+    const spy = hanbi.spy();
+
+    new MemoryCache(spy.handler).get('miss');
+    expect(spy.called).to.equal(true);
+    expect(spy.calledWith('miss')).to.equal(true);
+  });
+
+  it('should not call provided function when cache hit', function () {
+    const spy = hanbi.spy();
+
+    const memoryCache = new MemoryCache(spy.handler);
+    memoryCache.get('key');
+    spy.reset();
+
+    memoryCache.get('key');
+    expect(spy.called).to.equal(false);
+  });
+
+  it('should return the value returned by the provided function', function () {
+    const memoryCache = new MemoryCache(() => 'value');
+    const value = memoryCache.get('key');
+    expect(value).to.equal('value');
+  });
+
+  it('should return the right value', function () {
+    const memoryCache = new MemoryCache((key) => `${key}->value`);
+    expect(memoryCache.get('key1')).to.equal('key1->value');
+    expect(memoryCache.get('key2')).to.equal('key2->value');
+  });
+
+  it('should return the value found in cache', function () {
+    const memoryCache = new MemoryCache(() => 'value');
+    memoryCache.get('key');
+    const value = memoryCache.get('key');
+    expect(value).to.equal('value');
```

Comment:
This test is not about checking if the value is retrieved with a cache miss or a cache hit (which are tested with the `should call provided function when cache miss` and `should not call provided function when cache hit` unit tests).

It is only for testing that the right value is returned in case of a cache hit.

Maybe the test has the wrong name.